Oleksandr.Svidovyi/Task_5/src/main/java/com/hillel/task_5/servlet/ClientServlet.java

```diff
@@ -0,0 +1,165 @@
+import com.hillel.task_5.model.request_model.ClientGetRequestModel;
+import com.hillel.task_5.model.request_model.ClientPostRequestModel;
+import com.hillel.task_5.model.request_model.ClientPutRequestModel;
+import lombok.extern.slf4j.Slf4j;
+
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+@WebServlet(name = "ClientServlet", urlPatterns = {"/client"})
+@Slf4j
+public class ClientServlet extends HttpServlet {
+    private ClientRepository clientRepository;
+    private int currentId = 1;
+
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+        resp.setContentType("application/json");
+        Gson gson = new Gson();
+        clientRepository = ClientRepository.getInstance();
+
+        try {
+            ClientGetRequestModel requestModel = gson.fromJson(req.getReader(), ClientGetRequestModel.class);
+
+            if (requestModel.getClientId() != null && !requestModel.getClientId().equals("")) {
+                Integer id = Integer.decode(requestModel.getClientId());
+
+                if (id <= clientRepository.getClients().size()) {
+                    Client client = clientRepository.getClients().get(id);
+
+                    if (client != null) {
+                        resp.getWriter().println(gson.toJson(client));
+                    } else {
+                        resp.getWriter().println("Client not found.");
+                        resp.setStatus(404);
+                    }
+                } else {
+                    resp.getWriter().println("The client's id is not valid");
+                    resp.setStatus(400);
+                }
+            } else {
+                resp.getWriter().println("Please, enter client's name");
+                resp.setStatus(400);
+            }
+        } catch (Exception e) {
+            log.error("Exception while trying creating a client");
+            resp.getWriter().println("Something went wrong. Please, try again.");
+            resp.setStatus(400);
+        }
+    }
+
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+        resp.setContentType("application/json");
+        Gson gson = new Gson();
+        clientRepository = ClientRepository.getInstance();
+
+        try {
+            ClientPostRequestModel requestModel = gson.fromJson(req.getReader(), ClientPostRequestModel.class);
+
+            if (requestModel.getClientName() != null && !requestModel.getClientName().equals("")) {
+                Integer clientId = currentId;
+                Client client = new Client(clientId, requestModel.getClientName());
+                clientRepository.getClients().put(clientId, client);
```

Comment:
Don't use this logic.

Create new method like clientRepository.save(client)

Reply:
Done.